### DISCUSSION TITLE
Add a `Scroll` Animation

### DISCUSSION THREAD

**Discussion by icedcoffeeee** [ManimCommunity/manim#2801] [Suggestions and Proposals]
## Description of proposed feature

I find myself creating this new class for my projects, and thought if it might be useful for other users.

```py
class Scroll(AnimationGroup):
    def __init__(self, mob, target_mob, shift=UP):
        super().__init__(
            FadeIn(target_mob, shift=shift),
            FadeOut(mob, shift=shift),
        )
```

## How can the new feature be used?

```py
class Test(Scene):
    def construct(self):
        a = Text("Hello")
        b = Text("World")
        self.add(a)
        # self.play(FadeIn(b, shift=UP), FadeOut(a, shift=UP))
        self.play(Scroll(a, b))
        self.wait()
```
Output:

https://user-images.githubusercontent.com/83535735/138587001-0dfe1f8f-f7d1-49cb-92dc-ded9590fa1df.mp4

## Additional comments

I'm just not sure if this might be too simple of an addition for manim

**Comment by WampyCakes** [comment]
This looks like one of those things that belongs as a Transform. IIRC we have a lot of different Transforms that should be consolidated like Fades were.

**Comment by Darylgolden** [comment]
I would love this!

**Comment by behackl** [comment]
We do have a `FadeTransform`; maybe this could be implemented via a `shift` parameter for it?

In general, `FadeTransform` tries to be a bit clever: the default behavior is to have the submobjects of the starting mobject move towards the submobjects of the target mobject, so you will notice that `FadeTransform(a, b)` in your example will cause the letters to look as if they are moving.

Semantically, "different" behavior for specifying a shift still makes sense IMO: If `None` is passed (the default), the submobjects try to determine their shift target individually, whereas when `shift` is specified, both original and target mobject are shifted by the specified amount.

I think this would be a nice addition to `FadeTransform`.
